### DISCUSSION TITLE
Portable Configurations for Windows

### DISCUSSION THREAD

**Discussion by Stephen-Leibowitz** [pawelsalawa/sqlitestudio#4421] [Q&A]
There are instructions for a [Portable Configuration](https://github.com/pawelsalawa/sqlitestudio/wiki/Configuration_directory_location#portable-configuration).

But the program still creates and writes to a registry key:
HKEY_CURRENT_USER\Software\SalSoft

Is there a way for it to write to the application folder instead?

**Comment by pawelsalawa** [comment]
If I remember correctly, registry entries are created only by an installator. SQLiteStudio 3.3.3 does not have installator, thus should not create such entries.
Which version are you using?

**Comment by Stephen-Leibowitz** [comment]
I downloaded sqlitestudio-3.3.3.zip and extracted to the folder SQLiteStudio. I disabled the antivirus and closed other programs. I monitored with Regshot the program: 
Run > C:\PortableApps\SQLiteStudio\sqlitestudio.exe

**Reply by pawelsalawa** [reply]
And what are values that is creates? Not just keys/folders but are there any actual values?

**Comment by Stephen-Leibowitz** [comment]
Here is the value it wrote:
HKEY_CURRENT_USER\Software\SalSoft\SQLiteStudio\AllowMultipleSessions:
"false"

**Reply by pawelsalawa** [reply]
Ohh, right. Okay, that could be adjusted for portable distribution.

**Reply by pawelsalawa** [reply]
Created #4422 for it.

**Comment by John-foobar-Doe** [comment]
Same problem here, but even worse:

1. After extracting sqlitestudio-3.3.3.zip to `U:\PortableApps\SQLiteStudio\`, and launching the SQLiteStudio.exe via the PortableApps.com launcher, there is no `sqlitestudio-cfg/settings3` generated in `U:\PortableApps\SQLiteStudio\`, which is the expected behavior according to https://github.com/pawelsalawa/sqlitestudio/wiki/Configuration_directory_location#portable-configuration ;
2. The configuration is NOT saved in `%APPDATA%\SalSoft\sqlitestudio\settings3`, which for my case should be `C:\Users\Administrator\AppData\`**`Roaming`**`\SalSoft\sqlitestudio\settings3`, which is the expected behavior according to https://github.com/pawelsalawa/sqlitestudio/wiki/Configuration_directory_location#windows ;
3. Instead, the configuration is saved in `C:\Users\Administrator\AppData\`**`Local`**`\SalSoft\SQLiteStudio\settings3` ;
4. Same problem of generation of registry value `HKEY_CURRENT_USER\Software\SalSoft\SQLiteStudio\AllowMultipleSessions` which is `false`.

**Reply by Stephen-Leibowitz** [reply]
I think an earlier version of SQLiteStudio placed settings3 in the Roaming folder.

I previously used folder redirection to make SQLiteStudio portable on Windows. The technique is also usable with other applications. You may want to read my article about it at https://sleibowitz.neocities.org/FolderRedir.html

SQLiteStudio is a great application. I will be happy to publish a batch file and instructions to make it portable as soon as the registry issue is resolved.
